### DISCUSSION TITLE
v1.6.0

### DISCUSSION THREAD

**Discussion by Markemp** [Markemp/Cryengine-Converter#176] [Announcements]
- Support v3.22 Star Citizen Ivo files. (issue #168)
- Better material handling. (issue #161)
- Material bugfix for models that have 2 material files (Archeage). (issue #161)
- Allow the material file to be passed in via `-mtl` argument. This overrides the materials found in the `MtlNameChunk` which sometimes doesn't point to the right material file (MWO mech parts).
- Support for 0x0804 material chunks (New World). (issue #165)
- Support for 0x0801 compiled physical proxies chunk (Archeage fix). (issue #161, #74)
- Renamed all the Grendgine files to just Collada.
- Support for new bone list in Star Citizen files (issue #156)
- Star Citizen landing gear fix (issue #151)

<hr /><em>This discussion was created from the release <a href='https://github.com/Markemp/Cryengine-Converter/releases/tag/'>v1.6.0</a>.</em>

**Comment by giniedp** [comment]
I Had some time to try out the 1.6.0 release using gltf export and i am stuck on the following issue.

When the converter fails to read any texture for a material, the according meshes will have no UV coordinates at all.

It would be nice if the converter would pass through all geometry data without optimization, or is there such a mode already? I tend to use the converter purely for converting the geometry and do my own optimizations afterwards with other tools, so i need the data to be passed through without chopping.

Also, when passing -notex flag or -png flag, both have no effect. Converter still tries to convert textures and use the format given by material file. In my case the materials have tif files references but are actually dds files, which i pretransform to png files. 

```
[ERR:female_legendaryheavychest3_forearms_hands_ec5c7a991a2c3b5f4413ecd57725e90e.skin] Material[male_LegendaryHeavyChest3_00_chest_mat:0]: Failed to decode: objects/characters/player/male/mutatorlegendaryheavychest3/textures/male_mutatorlegendaryheavychest3_chest_ddna.tif
  IndexOutOfRangeException: Index was outside the bounds of the array.
     at BCnEncoder.Decoder.BcDecoder.DecodeInternal(DdsFile file, Boolean allMipMaps, CancellationToken token)
     at BCnEncoder.Decoder.BcDecoder.Decode(DdsFile file)
     at CgfConverter.Renderers.Gltf.BaseGltfRenderer.WriteMaterial(ChunkNode nodeChunk) in E:\Projects\github\Cryengine-Converter\CgfConverter\Renderers\Gltf\BaseGltfRenderer.Material.cs:line 131
```

@Markemp could you please have a look into it?

**Reply by Soreepeong** [reply]
Dropping UV when no texture is there was because Windows 3D Viewer didn't like having UV supplied without relevant textures; guess it's possible to output a blank texture instead instead of dropping UV.

For a quick fix to fit your purpose, modify `BaseGltfRenderer.Geometry.cs:298` and `:301` to be simply `= true;` instead of checking for textures.

**Reply by giniedp** [reply]
yeah, i'm already on it. Just running into another issue. The v1.6.0 release has a fix for https://github.com/Markemp/Cryengine-Converter/issues/165 but when switching and building the according commit, the fix is not there. Is there maybe a commit missing or somthing? Will poke further in a few hours.
